src/kubernetes.cc

```diff
@@ -961,6 +961,38 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidatePodConnectivity() const {
```

Comment:
[Optional] I feel like making those into separate functions is overkill - we don't intend to reuse them. Can we not just have blocks within `ValidateConfiguration` that do this?

Reply:
Done. I personally prefer smaller easily testable functions, but being that I'm not doing the relevant unit testing that goes with that statement, I'll streamline it.